### PR TITLE
Improve and simplify the EnrollmentManager in preparation for PreImageCache

### PR DESCRIPTION
**Pull request by Geod24**
While trying to adapt the EnrollmentManager to use PreImageCache (among other things), @linked0 and I made a few refactorings.

### CODE REVIEW

**Review by linked0 — APPROVED** [review]
LGTM. It's the output of the pair programming of @Geod24 & me.

File: source/agora/consensus/EnrollmentManager.d [hunk 1]
```diff
@@ -505,10 +505,7 @@ public class EnrollmentManager
```
Comment (AndrejMitrovic) [inline source/agora/consensus/EnrollmentManager.d:505]: This commit message has a typo `nedd`. :>

File: source/agora/consensus/EnrollmentManager.d [hunk 2]
```diff
@@ -659,18 +659,38 @@ public class EnrollmentManager
...
+        Params:
+            seed = The initial value for this round to derive pre-images from.
+            height = height at which the seed is.
+
+        Returns:
+            The last value for the cycle
+
+    ***************************************************************************/
+
+    public Hash populateCycleCache (Hash seed, ulong height) @safe nothrow
```
Comment (AndrejMitrovic) [inline source/agora/consensus/EnrollmentManager.d:679]: This is a `public` function on purpose? Because it doesn't have a unittest.

Comment (Geod24) [reply source/agora/consensus/EnrollmentManager.d:679]: Oops

**Pull request merged**